src/main/java/ch/njol/skript/expressions/ExprBannerPatterns.java

```diff
@@ -0,0 +1,271 @@
+			if (object instanceof Block block) {
+				if (block.getState() instanceof Banner banner) {
+					if (placement != null) {
+						if (banner.numberOfPatterns() >= placement)
+							patterns.add(banner.getPattern(placement - 1));
+					} else {
+						patterns.addAll(banner.getPatterns());
+					}
+				}
+			} else {
+				ItemStack itemStack = ItemUtils.asItemStack(object);
+				if (itemStack == null || !(itemStack.getItemMeta() instanceof BannerMeta bannerMeta))
+					continue;
+				if (placement != null) {
+					if (bannerMeta.numberOfPatterns() >= placement)
+						patterns.add(bannerMeta.getPattern(placement - 1));
+				} else {
+					patterns.addAll(bannerMeta.getPatterns());
+				}
+			}
+		}
+		return patterns.toArray(new Pattern[0]);
+	}
+
+	@Override
+	public Class<?> @Nullable [] acceptChange(ChangeMode mode) {
+		return switch (mode) {
+			case SET, DELETE -> {
+				if (patternNumber != null)
+					yield CollectionUtils.array(Pattern.class);
+				yield CollectionUtils.array(Pattern[].class);
+			}
+			case REMOVE, ADD -> CollectionUtils.array(Pattern[].class);
+			default -> null;
+		};
+	}
+
+	@Override
+	public void change(Event event, Object @Nullable [] delta, ChangeMode mode) {
+		Pattern[] patterns = null;
+		Pattern pattern = null;
+		Integer placement = null;
+		if (patternNumber != null) {
+			placement = patternNumber.getSingle(event);
+			if (delta != null && delta[0] != null) {
+				pattern = (Pattern) delta[0];
+			}
+		} else if (delta != null) {
+			patterns = (Pattern[]) delta;
+		}
+		Integer finalPlacement = placement;
+		Pattern finalPattern = pattern;
+		List<Pattern> patternList = patterns != null ? Arrays.stream(patterns).toList() : new ArrayList<>();
+
+		Consumer<BannerMeta> metaChanger = null;
+		Consumer<Banner> blockChanger = null;
+
+		if (placement != null) {
+			metaChanger = getPlacementMetaChanger(mode, finalPlacement, finalPattern);
+			blockChanger = getPlacementBlockChanger(mode, finalPlacement, finalPattern);
+		} else {
+			metaChanger = getAllMetaChanger(mode, patternList);
+			blockChanger = getAllBlockChanger(mode, patternList);
+		}
+
+		for (Object object : objects.getArray(event)) {
+			if (object instanceof Block block && block.getState() instanceof Banner banner) {
+				blockChanger.accept(banner);
+				banner.update(true, false);
+			} else {
+				ItemStack itemStack = ItemUtils.asItemStack(object);
+				if (itemStack == null || !(itemStack.getItemMeta() instanceof BannerMeta bannerMeta))
+					continue;
+				metaChanger.accept(bannerMeta);
+				itemStack.setItemMeta(bannerMeta);
+				if (object instanceof Slot slot) {
+					slot.setItem(itemStack);
+				} else if (object instanceof ItemType itemType) {
+					itemType.setItemMeta(bannerMeta);
+				} else if (object instanceof ItemStack itemStack1) {
+					itemStack1.setItemMeta(bannerMeta);
+				}
+			}
+		}
+
+	}
+
+	@Override
+	public boolean isSingle() {
+        return patternNumber != null && getExpr().isSingle();
+    }
```

Comment:
indentation (is your ide screwed up? why does this keep happening?)

Reply:
Not sure, usually happens when I cut/copy + paste.